orttraining/orttraining/eager/ort_aten.cpp

```diff
@@ -499,9 +499,20 @@ at::Tensor _copy_from_and_resize(
   const auto ort_self = create_ort_value(invoker, self);
   auto ort_dst = create_ort_value(invoker, dst);
 
+  // If the size of the dst is not equal to src, re-allocate the dst tensor to
+  // match the src tensor size.
+  if (self.numel() != dst.numel()) {
```

Comment:
Review: consider comparing src vs dst size in bytes.

Also - how does src vs dst types matter (can we assume the tensors will be of same type?) do we need to handle shape?